### DISCUSSION TITLE
CCCM and UART

### DISCUSSION THREAD

**Discussion by D4gu8** [Louisvdw/dbus-serialbattery#71] [Q&A] [answered]
Hey, Luis,
I've been looking for a good, working, isolated UART converter for a long time, but I haven't found one.  Finally I had to make it myself and already my LLT/JBD BMS is communicating. Do you recommend the original Victron ve-direct cable, the one that is isolated and at TTL level for use with the UART on the BMS? 
The second question is towards the SOC/current to battery (CCCM) setup. Where can these values be changed and how to enable this feature?

Thank you for the answer DJ.

**Comment by D4gu8** [comment]
Hi, so I found this in the battery.py script and after changing it, it's already working decently. I just want to play with it some more. Thanks for your work anyway.

**Comment by Louisvdw** [comment, marked as answer]
Hi @D4gu8 
The ve.Direct cable is an isolated cable which is a great safety feature. Howeven many people have problem if they cannot get the same connector that the BMS use to connect the cable to it, and this can cause signal loss or instability. If you do have the correct connector for your BMS and can make a solid connection, then the ve.Direct cable is great fr the extra safety. Else I would recommend you use the cable that came with the BMS for stability.

Just to confirm about CCCM, it is on by default. You can find the function that handles it in battery.py called **manage_charge_current**
